src/dd/dd.rs

```diff
@@ -0,0 +1,640 @@
+                },
+                "conv" => for elem in val.split(",") {
+                    match Conf::get_flag(elem) {
+                        Conf::INVALID_FLAG => err("invalid flag passed to conv"),
+                        flag => cf.set_flag(flag),
+                    }
+                },
+                inv => err(format!("invalid argument: {}", inv)),
+            }
+        } else {
+            err(format!("invalid argument: {}", arg));
+        }
+    }
+    cf
+}
+
+fn setup(cf: &mut Conf) {
+    let inp = if cf.is_stdin {
+        Some(InpFileWrap {
+            f: InpFileType::Stdin(io::stdin()), // TODO: use .lock() if necessary
+            is_seekable: false,
+        })
+    } else {
+        let f = File::open(cf.inp_name.as_ref() as &str);
+        if f.is_err() {
+            err("cannot open input file");
+            None
+        } else {
+            Some(InpFileWrap {
+                f: InpFileType::Reg(f.unwrap()),
+                is_seekable: true, // TODO: handle case when device isn't seekable
+            })
+        }
+    };
+    let mut inp = inp.unwrap();
+
+    let out = if cf.is_stdout {
+        Some(OutFileWrap {
+            f: OutFileType::Stdout(io::stdout()), // TODO: use .lock() if necessary
+            is_seekable: false,
+        })
+    } else {
+        let f = File::create(cf.out_name.as_ref() as &str);
+        if f.is_err() {
+            err("cannot open output file");
+            None
+        } else {
+            Some(OutFileWrap {
+                f: OutFileType::Reg(f.unwrap()),
+                is_seekable: true, // TODO: handle case when device isn't seekable
+            })
+        }
+    };
+    let mut out = out.unwrap();
+
+    match inp.f {
+        InpFileType::Reg(ref mut f) => {
+            if inp.is_seekable {
+                match f.seek(io::SeekFrom::Start((cf.skip * cf.inp_block_size) as u64)) {
+                    // .skip because inp is input
+                    Ok(_) => (),
+                    _ => err("cannot skip on input file"),
+                }
+            } else {
+                // TODO: implement by reading necessary amount of bytes from beginning
+            }
+        }
+        InpFileType::Stdin(ref s) => {
+            // TODO: implement by reading necessary amount of bytes from beginning
+        }
+    }
+
+    match out.f {
+        OutFileType::Reg(ref mut f) => {
+            if out.is_seekable {
+                match f.seek(io::SeekFrom::Start(cf.seek as u64)) {
+                    Ok(_) => (),
+                    _ => err("cannot seek on output file"),
+                }
+            } else {
+                // TODO: implement seeking on not seekable device
+            }
+        }
+        OutFileType::Stdout(ref s) => {
+            // TODO: implement seeking on not seekable device
+        }
+    }
+
+    let mut buf = vec![0u8; cmp::min(cf.inp_block_size, cf.out_block_size)];
```

Comment:
I would move the stuff from here onward into another function.  IMO in `uumain()` you should basically do something like:
```rust
let (input, output) = setup(&mut conf);
copy(input, output, &conf);
```

Reply:
Did.